observability/metrics/src/main/scala/com/daml/metrics/HealthMetrics.scala

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.metrics
+
+import com.daml.metrics.api.MetricHandle.Factory
+import com.daml.metrics.api.{MetricsContext, MetricName => MN}
+import java.util.concurrent.TimeoutException
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+class HealthMetrics(val factory: Factory) {
+  import HealthMetrics._
+
+  def registerHealthGauge(componentName: String, supplier: () => Boolean): Unit = {
+    val asLong = () => if (supplier()) 1L else 0L
+    factory.gaugeWithSupplier(MetricName, asLong, "The status of the Daml components")(
+      MetricsContext((ComponentLabel, componentName))
+    )
+  }
+
+  def registerHealthGauge(componentName: String, supplier: () => Future[Boolean])(implicit
+      executionContext: ExecutionContext
+  ): Unit = {
+    registerHealthGauge(
+      componentName,
+      // gaugeWithSupplier underlying code requires the value to be provided in a finite amount of time
+      // If the future has not been completed after a short timeout, force return false.
+      (() => {
+        try {
+          Await.result(supplier().fallbackTo(Future(false)), FutureSupplierTimeout)
```

Comment:
Yeah. I tried a few things.
There are documented ways to create future with timeouts based on the scala Futures, without blocking. But it would require to create a FutureWithTimeout type, with support for default in case of failure, to have a clean API.
It is a bit much if don't really need it.